src/puppetlabs/puppetdb/command/dlo.clj

```diff
@@ -71,31 +76,43 @@
                           (.printStackTrace exception out))
                         attempts))))
 
+(defn- strip-metadata-suffix
+  [queue-metadata-str]
+  (str/replace queue-metadata-str #"\.json$" ""))
```

Comment:
Fair enough - suppose I was mostly thinking of the constant factors i.e. length of string vs ~4, but in any case, it's not critical, and agreed that the inline subs isn't as clear; perhaps eventually something like this (with a better name?) in utils might be best of both worlds:
```
(defn chomp-string-suffix [s suffix]
  (if-let [n (str/last-index-of s suffix)]
    (subs s 0 n)
    s))
```